File: WikiCodeParser/Processors/INodeProcessor.cs
using System.Collections.Generic;
using WikiCodeParser.Nodes;

namespace WikiCodeParser.Processors
{
    public interface INodeProcessor
    {
        /// <summary>
        /// Higher priority processors are run first.
        /// </summary>
        int Priority { get; set; }

        /// <summary>
        /// Return true if the given node should be processed by this processor
        /// </summary>
        bool ShouldProcess(INode node, string scope);

        /// <summary>
        /// Process a node and return the nodes that will replace it.
        /// </summary>
        /// <returns>The nodes to replace the given node with</returns>
        IEnumerable<INode> Process(Parser parser, ParseData data, INode node, string scope);
    }
}

File: javascript/src/Processors/INodeProcessor.ts
import { INode } from '../Nodes/INode';
import { ParseData } from '../ParseData';
import { Parser } from '../Parser';

export interface INodeProcessor {
    /**
     * Higher priority processors are run first.
     */
    Priority : number;

    /**
     * Return true if the given node should be processed by this processor
     */
    ShouldProcess(node : INode, scope : string) : boolean;

    /**
     * Process a node and return the nodes that will replace it.
     * @returns The nodes to replace the given node with
     */
    Process(parser : Parser, data : ParseData, node : INode, scope : string) : INode[];
}
